src/PostgREST/ApiRequest/Preferences.hs

```diff
@@ -215,13 +215,13 @@ instance ToAppliedHeader PreferTransaction
 -- |
 -- How to handle the insertion/update when the keys specified in ?columns are not present
 -- in the json body.
-data PreferUndefinedKeys
-  = ApplyDefaults  -- ^ Use the default column value for the unspecified keys.
-  | IgnoreDefaults -- ^ Inserts: null values / Updates: the keys are not SET to any value
+data PreferMissing
+  = ApplyDefaults  -- ^ Use the default column value for missing values.
+  | ApplyNulls     -- ^ Use the null value for missing values. Inserts: null values / Updates: the keys are not SET to any value
```

Comment:
```suggestion
  | ApplyNulls     -- ^ Use the null value for missing values.
```
I wrongly added this here. For updates, this sets to `null` values and does not ignore them (I proposed to add another header value for those cases here: https://github.com/PostgREST/postgrest/pull/2693#discussion_r1148326675)